app/gui2/src/components/GraphEditor/GraphVisualization.vue

```diff
@@ -36,6 +39,62 @@ const emit = defineEmits<{
   setVisualizationVisible: [visible: boolean]
 }>()
 
+const visPreprocessor = ref(DEFAULT_VISUALIZATION_CONFIGURATION)
+const error = ref<Error>()
+
+const projectStore = useProjectStore()
+const graphStore = useGraphStore()
+const visualizationStore = useVisualizationStore()
+
+const expressionInfo = computed(
+  () => props.expressionId && graphStore.db.getExpressionInfo(props.expressionId),
+)
+const typeName = computed(() => expressionInfo.value?.typename ?? 'Any')
+
+/** The Visualization ID for the method to get the default visualization. */
+const defaultVisualizationVisualizationId = random.uuidv4() as Uuid
+const defaultVisualization = computedAsync(async () => {
+  if (props.currentType) return
+  if (!props.expressionId) return
+  const ident = props.pattern?.repr()
+  if (!ident) return
+  const ls = await projectStore.lsRpcConnection
+  const dataServer = await projectStore.dataConnection
+  ls.executeExpression(
+    projectStore.executionContext.id,
+    defaultVisualizationVisualizationId,
+    props.nodeId as ExprId,
+    `${ident}.default_visualization.to_js_object.to_json`,
+  )
+  return new Promise<VisualizationIdentifier | undefined>((resolve) => {
+    const onVisualizationUpdate = (payload: VisualizationUpdate, uuid: Uuid | null) => {
+      if (uuid !== defaultVisualizationVisualizationId) return
+      dataServer.off(`${OutboundPayload.VISUALIZATION_UPDATE}`, onVisualizationUpdate)
+      const data = payload.dataString()
+      const parsed: { library: { name: string } | null; name: string } | undefined = data
+        ? JSON.parse(data)
+        : undefined
+      resolve(
+        parsed && {
+          name: parsed.name,
+          module:
+            parsed.library == null
+              ? { kind: 'Builtin' }
+              : { kind: 'Library', name: parsed.library.name },
+        },
+      )
+    }
+    dataServer.on(`${OutboundPayload.VISUALIZATION_UPDATE}`, onVisualizationUpdate)
+  })
```

Comment:
You may copy `@/util/observable.ts` [from my PR](https://github.com/enso-org/enso/pull/8385) and use `nextEvent` here.